gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactionTask.java

```diff
@@ -61,17 +66,27 @@ public void onMRTaskComplete (boolean isSuccess, Throwable throwable) {
       try {
         List<CompactionCompleteAction> actions = this.suite.getCompactionCompleteActions();
         for (CompactionCompleteAction action: actions) {
+          action.addEventSubmitter(eventSubmitter);
           action.onCompactionJobComplete(dataset);
         }
+        submitEvent(CompactionSlaEventHelper.COMPACTION_COMPLETED_EVENT_NAME);
         super.onMRTaskComplete(true, null);
       } catch (IOException e) {
+        submitEvent(CompactionSlaEventHelper.COMPACTION_FAILED_EVENT_NAME);
         super.onMRTaskComplete(false, e);
       }
     } else {
+      submitEvent(CompactionSlaEventHelper.COMPACTION_FAILED_EVENT_NAME);
       super.onMRTaskComplete(false, throwable);
     }
   }
 
+  private void submitEvent(String eventName) {
+    Map<String, String> eventMetadataMap = new HashMap<>();
+    eventMetadataMap.put("datasetUrn", this.dataset.datasetURN());
+    this.eventSubmitter.submit(eventName, eventMetadataMap);
```

Comment:
You can use ImmutableMap<String,String>.of("datasetUrn", this.dataset.datasetURN()).
Also, a constant should be created for "datasetUrn" and the other references to "datasetUrn" above should be changed to use the constant.